src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -94,7 +107,23 @@ namespace Microsoft.Quantum.Convert {
     /// let bi2 = BoolArrayAsBigInt([false, false, false, false, false, false, false, true]); // Not padded -> -128
     /// ```
     function BoolArrayAsBigInt(a : Bool[]) : BigInt {
-        body intrinsic;
+        mutable val = 0L;
+        mutable arr = a;
+
+        if Length(arr) % 8 != 0 {
+            // Padding is needed.
+            set arr += [false, size = Length(arr) % 8];
+        }
```

Comment:
Currently I have an impression that if in L113 the `Length(arr)` is 6, then L115 will add **6** more (rather than **2** more), as a result the array will have length **12** (instead of **8**). Intuitively I feel that you meant this:
```suggestion
        if Length(arr) % 8 != 0 {
            // Padding is needed.
            set arr += [false, size = 8 - (Length(arr) % 8)];
        }
```

Reply:
Aha, good catch! My math was wrong there, glad you spotted it.